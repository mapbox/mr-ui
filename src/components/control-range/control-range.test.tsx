import React from 'react';
import ControlRange from './control-range';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('ControlRange', () => {
  describe('basic', () => {
    const mockOnChange = jest.fn();
    const props = {
      id: 'testinput',
      onChange: mockOnChange
    };

    test('renders', () => {
      const { baseElement } = render(<ControlRange {...props} />);
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('disabled', () => {
    const mockOnChange = jest.fn();
    const props = {
      id: 'testinput',
      onChange: mockOnChange,
      disabled: true
    };

    test('renders', () => {
      const { baseElement } = render(<ControlRange {...props} />);
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('all options', () => {
    const mockOnChange = jest.fn();
    const props = {
      id: 'testinput',
      label: 'All options',
      validationError: 'oh no!',
      value: [200, 400, 600],
      min: 100,
      max: 1000,
      step: 100,
      optional: true,
      autoFocus: true,
      aside: <span>Aside text</span>,
      themeControlRange: 'range--s range--red',
      themeControlRangeActive: 'bg-red-dark',
      themeControlWrapper: 'bg-red-light',
      themeLabel: 'txt-s txt-bold color-white',
      onChange: mockOnChange
    };

    test('renders', () => {
      const { baseElement } = render(<ControlRange {...props} />);
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('tooltip', () => {
    const mockOnChange = jest.fn();
    window.HTMLElement.prototype.hasPointerCapture = jest.fn();
    const props = {
      id: 'testinput',
      label: 'All options',
      validationError: 'oh no!',
      value: [200],
      min: 100,
      max: 1000,
      step: 100,
      optional: true,
      autoFocus: true,
      aside: <span>Aside text</span>,
      themeControlRange: 'range--s range--red',
      themeControlRangeActive: 'bg-red-dark',
      themeControlWrapper: 'bg-red-light',
      themeLabel: 'txt-s txt-bold color-white',
      tooltip: true,
      onChange: mockOnChange
    };
    test('renders', async () => {
      render(<ControlRange {...props} />);
      await userEvent.hover(screen.getByRole('slider'));

      await waitFor(() => {
        expect(screen.getByRole('tooltip')).toBeInTheDocument();
      });
    });
  });
});
