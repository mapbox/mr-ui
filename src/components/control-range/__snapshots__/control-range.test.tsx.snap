// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlRange all options renders 1`] = `
<body>
  <div>
    <div
      class="relative bg-red-light"
    >
      <div>
        <label
          class="inline-block txt-s txt-bold color-white"
          for="testinput"
        >
          All options
           
          <span
            class="txt-normal"
          >
            (optional)
          </span>
        </label>
        <span
          class="inline-block ml12"
        >
          <span>
            Aside text
          </span>
        </span>
      </div>
      <div
        class="range range--s range--red"
      >
        <span
          aria-disabled="false"
          aria-invalid="true"
          aria-required="false"
          data-orientation="horizontal"
          data-test="testinput-input"
          dir="ltr"
          id="testinput"
          style="--radix-slider-thumb-transform: translateX(-50%);"
        >
          <span
            data-orientation="horizontal"
          >
            <span
              class="absolute h-full bg-red-dark"
              data-orientation="horizontal"
              style="left: 11.11111111111111%; right: 44.44444444444445%;"
            />
          </span>
          <span
            style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(11.11111111111111% + 0px);"
          >
            <span
              aria-label="Value 1 of 3"
              aria-orientation="horizontal"
              aria-valuemax="1000"
              aria-valuemin="100"
              aria-valuenow="200"
              data-orientation="horizontal"
              data-radix-collection-item=""
              role="slider"
              style=""
              tabindex="0"
            />
          </span>
          <span
            style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(33.33333333333333% + 0px);"
          >
            <span
              aria-label="Value 2 of 3"
              aria-orientation="horizontal"
              aria-valuemax="1000"
              aria-valuemin="100"
              aria-valuenow="400"
              data-orientation="horizontal"
              data-radix-collection-item=""
              role="slider"
              style=""
              tabindex="0"
            />
          </span>
          <span
            style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(55.55555555555555% + 0px);"
          >
            <span
              aria-label="Value 3 of 3"
              aria-orientation="horizontal"
              aria-valuemax="1000"
              aria-valuemin="100"
              aria-valuenow="600"
              data-orientation="horizontal"
              data-radix-collection-item=""
              role="slider"
              style=""
              tabindex="0"
            />
          </span>
        </span>
      </div>
      <div
        role="alert"
      >
        <div
          class="bg txt-s mt3 py6 px12 bg-red-faint round border border--red color-red"
          data-testid="testinput-error"
          id="testinput-error"
        >
          oh no!
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`ControlRange basic renders 1`] = `
<body>
  <div>
    <div
      class="relative "
    >
      <div
        class="range range--s range--gray w-full h-full"
      >
        <span
          aria-disabled="false"
          aria-required="true"
          data-orientation="horizontal"
          data-test="testinput-input"
          dir="ltr"
          id="testinput"
          style="--radix-slider-thumb-transform: translateX(-50%);"
        >
          <span
            data-orientation="horizontal"
          >
            <span
              class="absolute h-full undefined"
              data-orientation="horizontal"
              style="left: 0%;"
            />
          </span>
        </span>
      </div>
      <div
        role="alert"
      />
    </div>
  </div>
</body>
`;

exports[`ControlRange disabled renders 1`] = `
<body>
  <div>
    <div
      class="relative "
    >
      <div
        class="range range--s range--gray w-full h-full"
      >
        <span
          aria-disabled="true"
          aria-required="true"
          data-disabled=""
          data-orientation="horizontal"
          data-test="testinput-input"
          dir="ltr"
          id="testinput"
          style="--radix-slider-thumb-transform: translateX(-50%);"
        >
          <span
            data-disabled=""
            data-orientation="horizontal"
          >
            <span
              class="absolute h-full undefined"
              data-disabled=""
              data-orientation="horizontal"
              style="left: 0%;"
            />
          </span>
        </span>
      </div>
      <div
        role="alert"
      />
    </div>
  </div>
</body>
`;
